code/modules/mining/lavaland/ash_flora.dm

```diff
@@ -0,0 +1,205 @@
+	luminosity = 1
+	harvested_name = "tiny mushrooms"
+	harvested_desc = "A few tiny mushrooms around larger stumps. You can already see them growing back."
+	harvest = /obj/item/weapon/reagent_containers/food/snacks/ash_flora/mushroom_stem
+	harvest_amount_high = 4
+	harvest_time = 40
+	harvest_message_low = "You pick and slice the cap off of a mushroom, leaving the stem."
+	harvest_message_med = "You pick and decapitate several mushrooms for their stems."
+	harvest_message_high = "You acquire a number of stems from these mushrooms."
+	regrowth_time_low = 3000
+	regrowth_time_high = 6000
+
+/obj/structure/flora/ash/cacti
+	icon_state = "cactus"
+	name = "fruiting cacti"
+	desc = "Several prickly cacti, brimming with ripe fruit and covered in a thin layer of ash."
+	harvested_name = "cacti"
+	harvested_desc = "A bunch of prickly cacti. You can see fruits slowly growing beneath the covering of ash."
+	harvest = /obj/item/weapon/reagent_containers/food/snacks/ash_flora/cactus_fruit
+	needs_sharp_harvest = FALSE
+	harvest_amount_high = 2
+	harvest_time = 10
+	harvest_message_low = "You pick a cactus fruit."
+	harvest_message_med = "You pick several cactus fruit." //shouldn't show up, because you can't get more than two
+	harvest_message_high = "You pick a pair of cactus fruit."
+	regrowth_time_low = 4800
+	regrowth_time_high = 7200
+
+/obj/structure/flora/ash/cacti/Crossed(mob/AM)
+	if(ishuman(AM) && has_gravity(loc) && prob(70))
+		var/mob/living/carbon/human/H = AM
+		if(!H.shoes && !H.lying) //ouch, my feet.
+			var/picked_def_zone = pick("l_leg", "r_leg")
+			var/obj/item/organ/external/affected = H.get_organ(picked_def_zone)
+			if(!istype(affected))
+				return
+			H.apply_damage(rand(3, 6), BRUTE, picked_def_zone)
+			H.Weaken(2)
+			H.visible_message("<span class='danger'>[H] steps on a cactus!</span>", \
+				"<span class='userdanger'>You step on a cactus!</span>")
+
+/obj/item/weapon/reagent_containers/food/snacks/ash_flora
+	name = "mushroom shavings"
+	desc = "Some shavings from a tall mushroom. With enough, might serve as a bowl."
+	icon = 'icons/obj/lavaland/ash_flora.dmi'
+	icon_state = "mushroom_shavings"
+	list_reagents = list("sugar" = 3, "ethanol" = 2, "stabilizing_agent" = 3, "minttoxin" = 2)
+	w_class = 1
+
+/obj/item/weapon/reagent_containers/food/snacks/ash_flora/New()
+	..()
+	pixel_x = rand(-4, 4)
+	pixel_y = rand(-4, 4)
+
+/obj/item/weapon/reagent_containers/food/snacks/ash_flora/shavings //for actual crafting
+
+/obj/item/weapon/reagent_containers/food/snacks/ash_flora/mushroom_leaf
+	name = "mushroom leaf"
+	desc = "A leaf, from a mushroom."
+	list_reagents = list("nutriment" = 3, "vitfro" = 2, "nicotine" = 2)
+	icon_state = "mushroom_leaf"
+
+/obj/item/weapon/reagent_containers/food/snacks/ash_flora/mushroom_cap
+	name = "mushroom cap"
+	desc = "The cap of a large mushroom."
+	list_reagents = list("mindbreaker" = 2, "entpoly" = 4, "mushroomhallucinogen" = 2)
```

Comment:
lsd instead of mindbreaker and psilocybin over mushroomhallucinogen...we don't have `entpoly` either.